src/stream.h

```diff
@@ -48,22 +47,24 @@ namespace librealsense
         rs2_stream _type = RS2_STREAM_ANY;
     };
 
+    namespace platform {
+        struct stream_profile;
+    }
+
     class backend_stream_profile
     {
     public:
-        explicit backend_stream_profile(platform::stream_profile sp) : _sp(std::move(sp)) {}
-
-        platform::stream_profile get_backend_profile() const { return _sp; }
+        virtual platform::stream_profile const & get_backend_profile() const {
+            throw not_implemented_exception( "not a backend profile!" );
+        }
 
         virtual ~backend_stream_profile() = default;
-    private:
-        platform::stream_profile _sp;
     };
 
     class stream_profile_base : public virtual stream_profile_interface, public backend_stream_profile
```

Comment:
Does `stream_profile_base` need to derive `backend_stream_profile`?
i.e. are all `video_stream_profile`s a `backend_stream_profile`?

Reply:
I meant that I think they don't necessarily have to be.
For next PR :-)